controllers/util/util.go

```diff
@@ -122,3 +126,43 @@ func GetSystemCerts() ([]byte, error) {
 	}
 	return data, err
 }
+
+func GetServiceHostname(ctx context.Context, svcName, ns string, client client.Client) (string, error) {
+	service := &v1.Service{}
+	namespacedNamed := types.NamespacedName{Name: svcName, Namespace: ns}
+	err := client.Get(ctx, namespacedNamed, service)
+	if err != nil {
+		return "", err
+	}
+
+	// Loop over all Service ports, if a secured port is found
+	// set port and scheme to its secured ones and skip the loop
+	servicePort := ""
+	serviceScheme := "http"
+	for port := range service.Spec.Ports {
+		servicePort = fmt.Sprintf("%d", port)
+		if servicePort == "8443" || servicePort == "443" {
+			// If a secured port is found, just set scheme to 'https://' and skip the loop
```

Comment:
```suggestion
			// If a secured port is found, just set scheme to 'https://' and exit the loop
```

("skip" means "never enter it", whereas "exit" means "leave it early")